tests/v2/test_waku_rln_relay.nim

```diff
@@ -628,9 +632,79 @@ suite "Waku rln relay":
 
     # verify the proof (should not be verified)
     let verified = rln.proofVerify(data = messageBytes,
-                                 proof = proof)
+                                  proof = proof)
+    check:
+      verified.isOk()
+      verified.value() == false
+
+  test "invalidate messages with a valid, but stale root":
```

Comment:
When designing a test, you should try to follow this structure:

* **Setup:** Setting up the test harness (e.g., async runtime, waku node). This step is optional.
* **Given:** You declare all the pieces that will interact in the test. These are the stimulus (e.g., test content topic string, prepopulated Merkle tree) and the SUT (System Under Test).
* **When:** Then you stimulate the SUT. Execute the action that you want to test and keep the system outputs.
* **Then:** Here, you do the assertions of the system outputs.
* **Cleanup/Teardown:** The last step is to clean up everything that was set up in the first stage; or restore the state for something that got mutated. This step is also optional.

This will make a huge difference in the quality of the code, as you'll start thinking about doing more testable code, and also, it will make the tests way easier to understand, review and maintain.

Reply:
As a rule of thumb, `check()` can only go in the **Then** section. And if there is a condition _sine qua non_ the test should halt, then you should use `require` that finishes the test execution. This second method can be used in any test stage. It is not harmful to use `require` in the _Then_ section, but in that section, it is preferable to just use `check()`.

Reply:
Addressed in [f75eef3](https://github.com/status-im/nwaku/pull/1158/commits/f75eef32439c50edb2b8395da8b3f3d425abe110)